src/scripts/main.js

```diff
@@ -1,3 +1,31 @@
 'use strict';
 
-// write code here
+const listElement = document.querySelector('ul');
+const workersArr = [...listElement.children];
+
+listElement.innerHTML = '';
+
+sortWorkers(workersArr);
+mapWorkers(workersArr);
+
+function sortWorkers(arr) {
+  const sortedWorkers = arr.sort((worker1, worker2) => {
+    const salary1 = +worker1.dataset.salary.replace(/[$,]/g, '');
+    const salary2 = +worker2.dataset.salary.replace(/[$,]/g, '');
+
+    return salary2 - salary1;
+  });
+
+  listElement.append(...sortedWorkers);
+}
+
+function mapWorkers(arr) {
+  return arr.map((worker) => {
+    return {
+      name: worker.innerText,
+      position: worker.dataset.position,
+      salary: worker.dataset.salary,
+      age: worker.dataset.age,
+    };
+  });
+}
```

Comment:
The `mapWorkers` function is correctly mapping the worker elements to objects with the required properties. However, the task requires a `getEmployees` function that should return this array of employee objects. Consider renaming or creating a new function to fulfill this requirement.